### ISSUE TITLE
ALE don't show some error from golangci-lint

### ISSUE THREAD

**Issue by liaol** [dense-analysis/ale#4526]
## Information

**VIM version**

NVIM v0.9.0
Build type: Release
LuaJIT 2.1.0-beta3

Operating System: MacOS 12.5.1 Intel

## What went wrong

golangci-lint exit with exitcode 1 and error， but ALE don't show in quicklist or line 

## Reproducing the bug
go code 

```
package main

func main() {
    abc()
}

func abc(a int) int {
    return a + 1
}
```

### :ALEInfo
<details>
  <summary>Expand</summary>
  
 Current Filetype: go
Available Linters: ['bingo', 'cspell', 'gobuild', 'gofmt', 'golangci-lint', 'golint', 'gometalinter', 'gopls', 'gosimple', 'gotype', 'govet', 'golangserver', 'revive', 'staticcheck']
   Linter Aliases:
'gobuild' -> ['go build']
'govet' -> ['go vet']
  Enabled Linters: ['golangci-lint']
  Ignored Linters: []
 Suggested Fixers: 
  'gofmt' - Fix Go files with go fmt.
  'gofumpt' - Fix Go files with gofumpt, a stricter go fmt.
  'goimports' - Fix Go files imports with goimports.
  'golines' - Fix Go file long lines with golines
  'gopls' - Fix Go files with gopls.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 Linter Variables:

let g:ale_go_go_executable = 'go'
let g:ale_go_goimports_executable = 'goimports'
let g:ale_go_goimports_options = ''
let g:ale_go_golangci_lint_executable = 'golangci-lint'
let g:ale_go_golangci_lint_options = ''
let g:ale_go_golangci_lint_package = 1
let g:ale_go_langserver_executable = 'go-langserver'
let g:ale_go_langserver_options = ''
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_disable_lsp = 0
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 1
let g:ale_fixers = {'go': ['goimports'], 'markdown': ['remark-lint'], '*': ['remove_trailing_lines', 'trim_whitespace'], 'python': ['black', 'autoimport', 'isort']}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 0
let g:ale_lint_on_filetype_changed = 0
let g:ale_lint_on_insert_leave = 0
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'never'
let g:ale_linter_aliases = {}
let g:ale_linters = {'go': ['golangci-lint'], 'markdown': ['remark-lint'], 'lua': ['luacheck -g'], 'html': [''], 'python': ['pylint']}
let g:ale_linters_explicit = 0
let g:ale_linters_ignore = {}
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 1
let g:ale_pattern_options = {'/usr/local/Cellar/go/.*\.go$': {'ale_enabled': 0}, '/usr/local/go/.*\.go$': {'ale_enabled': 0}}
let g:ale_pattern_options_enabled = v:null
let g:ale_root = {}
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_neovim_diagnostics_api = 0
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 'all'
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(finished - exit code 0) ['/bin/zsh', '-c', '''goimports'' -l -w -srcdir ''/private/tmp/abc/abc.go'' ''/var/folders/p8/vclzvkl51_j_k97kldgry8l00000gn/T/nvim.liao/CoDk3Z/0/abc.go''']
(executable check - success) golangci-lint
(finished - exit code 1) ['/bin/zsh', '-c', 'cd ''/private/tmp/abc'' && ''golangci-lint'' run ']

<<<OUTPUT STARTS>>>
abc.go:1: : # abc
./abc.go:4:2: not enough arguments in call to abc
	have ()
	want (int) (typecheck)
package main
<<<OUTPUT ENDS>>>

</details>

**Comment by pbnj** [comment]
Thank you for the bug report. I was able to reproduce this.

It seems like the regex pattern [here](https://github.com/dense-analysis/ale/blob/29e5ffe35a2fa1ca5ba19eebfe7cdd0191c16e39/ale_linters/go/golangci_lint.vim#L27) is the problem.

I pushed a PR to fix this after validating the new regex pattern works.

**Comment by w0rp** [comment]
This should be fixed in #4917, as we now load JSON output from the linter instead of matching output with regex.
